src/components/TextField/TextField.tsx

```diff
@@ -100,19 +98,26 @@ export default class TextField extends React.PureComponent<TextFieldProps, {}> {
         suffix={this.props.suffix}
         disabled={this.props.disabled}
         required={this.props.required}
+        errorMessage={this.props.errorMessage}
         placeholder={this.props.placeHolder}
         multiline={!!this.props.autoAdjustHeight || !!this.props.rows && (this.props.rows > 1)}
         rows={this.props.rows}
-        onGetErrorMessage={this.props.getErrorMessage}
-        deferredValidationTime={deferredValidationTime}
         resizable={false}
         autoAdjustHeight={this.props.autoAdjustHeight}
         underlined={this.props.underlined}
-        onChanged={this.props.onChanged}
+        onChanged={this.getOnChange()}
       />
     );
   }
 
+  private getOnChange() {
```

Comment:
we're essentially executing this method every time props change (which would generally be after each keystroke in a controlled component). i think i'd prefer to pass a function directly to onChanged instead of executing this get function on every render cycle.

Reply:
Thanks @swese44 , I was wondering about that too, and could probably benefit from some guidance here. Here are some alternatives I see:

1. Simply move this logic to the constructor. Downside is that if for some reason the consumer wants to change the `onChange` or `onChangeDebounceTime` props, it won't be possible. 

2. In addition to #1, apply this logic in componentWillUpdate, behind a check to see if either `onChange` or `onChangeDebounceTime` have changed.

Reply:
Actually... disregard this, I think I was overcomplicating things, and understand what you mean now.

Reply:
Scratch that, I'm back to where I was before. 
Do you think it's worth the complexity to do something like this?  Which part of the function looks the most costly to you?  Might some sort of perf test help us decide?

```js
  private onChange: (newValue: string) => void;

  constructor(props: TextFieldProps) {
    super();
    this.async = new Async(this);
    if (props.onChangeDebounceTime && props.onChange) {
      this.onChange = this.async.debounce(props.onChange, props.onChangeDebounceTime);
    }
  }

  componentWillUpdate(nextProps: TextFieldProps) {
    if (nextProps.onChangeDebounceTime && nextProps.onChange
      && (nextProps.onChangeDebounceTime !== this.props.onChangeDebounceTime
        || nextProps.onChange !== this.props.onChange)) {
      this.onChange = this.async.debounce(nextProps.onChange, nextProps.onChangeDebounceTime);
    }
  }

  render() {
    return (
      <FabricTextField
        value={this.props.value}
        className={this.getClasses()}
        label={this.props.label}
        description={this.props.description}
        prefix={this.props.prefix}
        suffix={this.props.suffix}
        disabled={this.props.disabled}
        required={this.props.required}
        errorMessage={this.props.errorMessage}
        placeholder={this.props.placeHolder}
        multiline={!!this.props.autoAdjustHeight || !!this.props.rows && (this.props.rows > 1)}
        rows={this.props.rows}
        resizable={false}
        autoAdjustHeight={this.props.autoAdjustHeight}
        underlined={this.props.underlined}
        onChanged={this.onChange}
      />
    );
  }
```

Reply:
this looks like the way Fabric's textfield does it, though they don't worry about updating a new update value so this is slightly improved. does this work as expected when providing a value (controlled component) and a debounce value?

Reply:
would probably also need to clear the async timeout on `componentWillUnmount`

Reply:
which looks to be `async.dispose()`:
https://github.com/OfficeDev/office-ui-fabric-react/blob/4b1468da53a1c3f5e0ddf5750b31c8500237ce6b/packages/utilities/src/Async.ts#L34